.github/workflows/publish.yaml

```diff
@@ -65,3 +78,33 @@ jobs:
             labels: ${{ steps.meta.outputs.labels }}
             cache-from: type=gha
             cache-to: type=gha,mode=max
+
+        # Sign the resulting Docker image digest except on PRs.
+        # This will only write to the public Rekor transparency log when the Docker
+        # repository is public to avoid leaking data.  If you would like to publish
+        # transparency data even for private images, pass --force to cosign below.
+        # https://github.com/sigstore/cosign
+        - name: Sign the published Docker image
+          # This step uses the identity token to provision an ephemeral certificate
+          # against the sigstore community Fulcio instance.
+          env:
+            TAGS: ${{ steps.meta.outputs.tags }}
+            DIGEST: ${{ steps.build-and-push.outputs.digest }}
+          run: echo "${TAGS}" | xargs -I {} cosign sign --yes {}@${DIGEST}
+    
+    provenance-for-images:
+      needs: [paranoia-inception]
+      permissions:
+        actions: read # for detecting the Github Actions environment.
+        id-token: write # for creating OIDC tokens for signing.
+        packages: write # for uploading attestations.
+      uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.6.0
```

Comment:
note: you can use https://github.com/slsa-framework/slsa-verifier to verify provenance, both for containers and binaries